blocks/rich-text/index.js

```diff
@@ -558,6 +560,21 @@ export class RichText extends Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onChange();
 		}
+
+		this.scrollToCaret();
+	}
+
+	scrollToCaret() {
+		const caretRect = this.getEditorSelectionRect();
+		const caretHeight = caretRect.y;
+		if ( caretHeight !== this.caretHeight ) {
+			const toolbarOffset = 100;
+			window.scrollTo(
+				window.pageXOffset,
+				window.pageYOffset + caretHeight - toolbarOffset
+			);
```

Comment:
`scrollIntoView` was behaving quite awkwardly for me, scrolling a bit unpredictably. I'd still rather use whatever our preferred interface is, at least one that looks at scrollable containers. For the sake of having something demoable I pushed this anyway (and there's something nice about just calling `window.scrollTo`).